semantic_search/main.py

```diff
@@ -80,7 +82,10 @@ def app_startup():
         settings.pretrained_model_name_or_path, cuda_device=settings.cuda_device
     )
     embedding_dim = model.model.config.hidden_size
-    model.index = setup_faiss_index(embedding_dim)
+    if settings.file_path is not None:
+        model.index = read_index(settings.file_path)
```

Comment:
```suggestion
        model.index = faiss.swigfaiss.read_index(settings.file_path)
```

This is more readable imo